content/docs/setup/kubernetes/spec-requirements/index.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Spec requirements
+description:  Pod and Services specification requirements
+weight: 99
+keywords: [kubernetes,sidecar,sidecar-injection]
+---
+
+## Pods and Services specification requirements
+
+To be a part of the service mesh, pods and services in the Kubernetes
+cluster must satisfy the following requirements:
+
+1. _**Named ports**:_ Service ports must be named. The port names must be of
+  the form `<protocol>[-<suffix>]` with _http_, _http2_, _grpc_, _mongo_, or _redis_
+  as the `<protocol>` in order to take advantage of Istio's routing features.
+  For example, `name: http2-foo` or `name: http` are valid port names, but
+  `name: http2foo` is not.  If the port name does not begin with a recognized
+  prefix or if the port is unnamed, traffic on the port will be treated as
+  plain TCP traffic (unless the port explicitly uses `Protocol: UDP` to
+  signify a UDP port).
+
+1. _**Service association**:_ If a pod belongs to multiple
+  [Kubernetes Service](https://kubernetes.io/docs/concepts/services-networking/service/),
+  the services cannot use the same port number for different protocols, for instance HTTP and TCP.
+
+1. _**Deployments with app label**:_ It is recommended that Pods deployed using
+  the Kubernetes `Deployment` have an explicit `app` label in the
+  Deployment specification. Each deployment specification should have a
+  distinct `app` label with a value indicating something meaningful. The
+  `app` label is used to add contextual information in distributed
```

Comment:
I read this sentence, and don't quite understand what it means. Can you provide something a bit more descriptive?  Where people will get hung up on this is `contextual information` imo.